### PR TITLE
Feature/1563 Make save buttons disable on click and re-enable after process is complete

### PR DESCRIPTION
**Pull request by sasha-dresden**
Also added a bunch of tests to pass the 80% mark

### CODE REVIEW

File: front-end/src/app/shared/components/navigation-control/navigation-control.component.html [hunk 1]
```diff
@@ -3,12 +3,11 @@
...
-    appSingleClick
     [label]="navigationControl?.label || ''"
     [icon]="navigationControl?.icon || ''"
     [class]="navigationControl?.ngClass || ''"
     (click)="click()"
-    [disabled]="isDisabled()"
+    [disabled]="isDisabled() || processing"
   ></button>
   <p-dropdown
     *ngIf="controlType === 'dropdown'"
```
Comment (toddlees) [inline front-end/src/app/shared/components/navigation-control/navigation-control.component.html:14]: we'll want to disable this dropdown as well.  It allows the user to save and add something from the list

File: front-end/src/app/shared/components/transaction-type-base/triple-transaction-type-base.component.ts [hunk 2]
```diff
@@ -119,47 +120,21 @@ export abstract class TripleTransactionTypeBaseComponent
...
-    } else {
-      this.navigateTo(navigationEvent);
-    }
+  override isInvalid(): boolean {
+    return super.isInvalid() ||
+      this.childForm_2.invalid ||
+      !this.childTransaction_2;
+  }
+
+  override get confirmation$(): Observable<boolean> {
```
Comment (toddlees) [inline front-end/src/app/shared/components/transaction-type-base/triple-transaction-type-base.component.ts:131]: WOW! This is a really impressive way of declaring the confirmations for each part once AND in their own component.

**Review by toddlees — CHANGES_REQUESTED** [review]

File: front-end/src/app/shared/components/navigation-control/navigation-control.component.html (hunk 1 above)
Comment (sasha-dresden) [reply front-end/src/app/shared/components/navigation-control/navigation-control.component.html:14]: Oh good catch. I added a commit to add the "or processing" to the dropdown as well.

**Review by toddlees — APPROVED** [review]

**Comment by sasha-dresden** [comment]
Updated code to make use of spinner store.
This reduced the need for passing the processing variable up and down the component hierarchy and also simplified the code.
I am now reusing the single click directive which has been updated to subscribe to the spinner store for updating the disable state.
I also update the ApiService::put and ::post methods so that there is a new Boolean variable it can accept which defaults to false, but if passed as true will use the Spinner variation.

The one potential issue with all of this is that the button gets disabled immediately on click, which is important to prevent double click submissions, however, if the button click doesn't result in an api call (for example if the form fails to validate) there are no calls to re-enable the button. In order to handle this situation, I allowed the button to re-enable itself through the directive after a delay, if the spinner status is false.

**Review by mjtravers — APPROVED** [review]

**Pull request merged**